src/legacy/server/config/schema.js

```diff
@@ -83,50 +81,33 @@ export default () => Joi.object({
   server: Joi.object({
     uuid: Joi.string().guid().default(),
     name: Joi.string().default(os.hostname()),
-    host: Joi.string().hostname().default('localhost'),
-    port: Joi.number().default(5601),
-    keepaliveTimeout: Joi.number().default(120000),
-    socketTimeout: Joi.number().default(120000),
-    maxPayloadBytes: Joi.number().default(1048576),
-    autoListen: Joi.boolean().default(true),
     defaultRoute: Joi.string().default('/app/kibana').regex(/^\//, `start with a slash`),
-    basePath: Joi.string().default('').allow('').regex(/(^$|^\/.*[^\/]$)/, `start with a slash, don't end with one`),
-    rewriteBasePath: Joi.boolean().when('basePath', {
-      is: '',
-      then: Joi.default(false).valid(false),
-      otherwise: Joi.default(false),
-    }),
     customResponseHeaders: Joi.object().unknown(true).default({}),
-    ssl: Joi.object({
-      enabled: Joi.boolean().default(false),
-      redirectHttpFromPort: Joi.number(),
-      certificate: Joi.string().when('enabled', {
-        is: true,
-        then: Joi.required(),
-      }),
-      key: Joi.string().when('enabled', {
-        is: true,
-        then: Joi.required()
-      }),
-      keyPassphrase: Joi.string(),
-      certificateAuthorities: Joi.array().single().items(Joi.string()).default([]),
-      supportedProtocols: Joi.array().items(Joi.string().valid('TLSv1', 'TLSv1.1', 'TLSv1.2')).default(['TLSv1.1', 'TLSv1.2']),
-      cipherSuites: Joi.array().items(Joi.string()).default(cryptoConstants.defaultCoreCipherList.split(':'))
-    }).default(),
-    cors: Joi.when('$dev', {
-      is: true,
-      then: Joi.object().default({
-        origin: ['*://localhost:9876'] // karma test server
-      }),
-      otherwise: Joi.boolean().default(false)
-    }),
     xsrf: Joi.object({
       disableProtection: Joi.boolean().default(false),
       whitelist: Joi.array().items(
         Joi.string().regex(/^\//, 'start with a slash')
       ).default([]),
       token: Joi.string().optional().notes('Deprecated')
     }).default(),
+
+    // keep them for BWC, remove when not used in Legacy.
+    // validation should be in sync with one in New platform.
```

Comment:
Can you add a file path for where these should be in sync?
Can you also add a comment in the corresponding NP code (unless there already is one)?

Reply:
> Can you add a file path for where these should be in sync?

ok. it's not always practical, because a file can be moved and comments are out of sync 

> Can you also add a comment in the corresponding NP code (unless there already is one)?

will do